app/services/NotificationService.scala

```diff
@@ -20,28 +20,52 @@ import connectors.{EmailConnector, EmailNotSent, EmailStatus}
 import org.joda.time.LocalDate
 import org.joda.time.format.DateTimeFormat
 import play.api.libs.json.JsValue
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.retrieve.Name
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.AuthFunctionality
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-import uk.gov.hmrc.http.HeaderCarrier
 
-trait NotificationService {
+trait NotificationService extends AuthFunctionality with Retrievals {
+
+  def extractName(authName: Option[Name]): String = {
+    authName match {
```

Comment:
`authName.flatMap(_.name).getOrElse("No first name present")`

I feel like this is a less complicated way to write what these functions do. Additionally I'd feel more comfortable in the fact they don't have tests if they were written in a shorter form

Reply:
Refactored into a flat map and moved into ated utils. Added unit tests as well.